text-generation-inference/Dockerfile

```
@@ -1,10 +1,17 @@
 # Fetch and extract the TGI sources (TGI_VERSION is mandatory)
 FROM alpine AS tgi
 ARG TGI_VERSION
-RUN test -n ${TGI_VERSION:?}
-RUN mkdir -p /tgi
-ADD https://github.com/huggingface/text-generation-inference/archive/refs/tags/v${TGI_VERSION}.tar.gz /tgi/sources.tar.gz
-RUN tar -C /tgi -xf /tgi/sources.tar.gz --strip-components=1
+RUN apk add git && \
+    test -n ${TGI_VERSION:?} && \
+    mkdir -p /tgi && \
+    cd /tgi && \
+    git init && \
+    git remote add origin https://github.com/huggingface/text-generation-inference && \
+    git fetch origin ${TGI_VERSION} && \
+    git reset --hard FETCH_HEAD
+
+#ADD https://github.com/huggingface/text-generation-inference/archive/refs/tags/v${TGI_VERSION}.tar.gz /tgi/sources.tar.gz
```

I think you could have just done
```suggestion
ADD https://github.com/huggingface/text-generation-inference/archive/${TGI_VERSION}.tar.gz
```

---

There is no tar.gz for specific commit AFAIK, is there?

---

yes it is available. It does not matter though, we might do this later.